src/scenario/action_types/action_types.c

```diff
@@ -282,6 +285,43 @@ int scenario_action_type_savings_add_execute(scenario_action_t *action)
     return 1;
 }
 
+int scenario_action_type_building_force_collapse_execute(scenario_action_t *action)
+{
+    int grid_offset = action->parameter1;
+    int block_radius = action->parameter2;
+    building_type type = action->parameter3;
+    int destroy_all = action->parameter4;
+
+    if (!map_grid_is_valid_offset(grid_offset)) {
+        return 0;
+    }
+
+    for (int y = -block_radius; y <= block_radius; y++) {
+        for (int x = -block_radius; x <= block_radius; x++) {
+            int current_grid_offset = map_grid_add_delta(grid_offset, x, y);
+            if (!map_grid_is_valid_offset(current_grid_offset)) {
+                continue;
+            }
+            int building_id = map_building_at(current_grid_offset);
+            if (!building_id) {
+                continue;
+            }
+            building *b = building_main(building_get(building_id));
+            if (b->type == BUILDING_BURNING_RUIN) {
+                continue;
+            }
+            if (b->state == BUILDING_STATE_DELETED_BY_PLAYER || b->is_deleted) {
```

Comment:
I think this would be preferrable:

```suggestion
            if ((b->state != BUILDING_STATE_IN_USE && b->state != BUILDING_STATE_MOTHBALLED) || b->is_deleted) {
```

Reply:
Fixed.